service/src/test/java/cz/muni/fi/pa165/mushrooms/service/MushroomHunterServiceTest.java

```diff
@@ -0,0 +1,374 @@
+                        if (hunter.getUserNickname().equals(nickname)){
+                            return hunter;
+                        }
+                    }
+                    return null;
+                }
+            };
+            minTimes = 0;
+
+            mushroomHunterDao.findAll();
+            result = new Delegate() {
+                List<MushroomHunter> foo() {
+                    return Collections.unmodifiableList(new ArrayList<>(persistedMushroomHunter.values()));
+                }
+            };
+            minTimes = 0;
+
+
+            mushroomHunterDao.delete((MushroomHunter) any);
+            result = new Delegate() {
+                void foo(MushroomHunter mushroomHunter) {
+                    if (mushroomHunter == null || mushroomHunter.getId() == null ){
+                        throw new IllegalArgumentException("invalid entity");
+                    }
+                    if (persistedMushroomHunter.get(mushroomHunter.getId()) == null){
+                        throw new IllegalArgumentException("not in db");
+                    }
+                    persistedMushroomHunter.remove(mushroomHunter.getId());
+                }
+            };
+            minTimes = 0;
+
+            mushroomHunterDao.update((MushroomHunter) any);
+            result = new Delegate() {
+                void foo(MushroomHunter mushroomHunter) {
+                    if (mushroomHunter == null){
+                        throw new IllegalArgumentException("null mushroom hunter on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getId() == null){
+                        throw new IllegalArgumentException("null id on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getFirstName() == null){
+                        throw new IllegalArgumentException("null first name on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getSurname() == null){
+                        throw new IllegalArgumentException("null surname on mushroom hunter update");
+                    }
+                    if (mushroomHunter.getUserNickname() == null){
+                        throw new IllegalArgumentException("null nickname on mushroom hunter update");
+                    }
+                    if (TestUtils.checkMushroomHunterDuplicity(persistedMushroomHunter, mushroomHunter)){
+                        throw new IllegalArgumentException("duplicating unique mushroom hunter attribute on updating");
+                    }
+                    persistedMushroomHunter.replace(mushroomHunter.getId(), mushroomHunter);
+                }
+            };
+            minTimes = 0;
+
+        }};
+
+        mushroomHunterDao.create(persistedHunter);
+        mushroomHunterDao.create(persistedHunter2);
+    }
+
+    @Test
+    public void findMushroomHunterById() throws Exception {
+        mushroomHunterDao.create(newHunter);
```

Comment:
the hunter should be created here (using TestUtils), so it's not unnecessarily created in setup for every test method